source-code/sdk-js/src/adapter-sveltekit/vite-plugin/transforms/refactoring.md

```diff
@@ -113,6 +113,144 @@ for markup block:
+				importFrom = "@inlang/sdk-js/adapter-sveltekit/server"
+				break
+			case "initActionWrapper":
+				identifiers = ["actions"]
+				wrapperAst = parse(`${wrapper}(/*options here*/).wrap`)
+				importFrom = "@inlang/sdk-js/adapter-sveltekit/server"
+				break
+			default:
+				break
+		}
+		// Wrap the specified functions
+		const wasWrapped = definitions(ast, ...identifiers)
+			.wrap(wrapperAst)
+			.successful()
+		if (wasWrapped) imports(ast, importFrom).add(wrapper)
+
+		// Add imports
+	})
+	const aliases = imports(ast, "@inlang/sdk-js").getAliases()
+	// Ensure that function parameters are correct
+	contexts(ast, ...aliases.map(({ aliasN }) => aliasN))
+		.isFunction()
+		.merge(
+			parse(
+				`function(_, {${aliases
+					.map(({ exportN, aliasN }) => `${exportN}:${aliasN}`)
+					.join(",")}}) {}`,
+			),
+		)
+	aliases.forEach(({ exportN, aliasN }) =>
+		// Wrap all occurences of `i`, `language`, etc with `get()`
+		identifiers(ast, aliasN).wrap(parse("get")),
+	)
+}
+
+function replaceSdkImports(ast) {
+	// Prunes the imports, i.e. merge duplicates.
+	// Then remove sdk imports, but save the aliases: import {i as iAlias} ... returns [{exportN: string, aliasN: string}, ...]
+	const aliases = imports(ast, "@inlang/sdk-js").prune().remove().getAliases()
+	// Declare all aliases from sdk-import: `let iAlias`, `let languageAlias`, ...
+	// Assign `({ i } = getRuntimeFromContext())` in each context of iAlias (or $iAlias in reactive case).
+	contexts(ast).insertAfterImports(
+		parse(`
+      $$_INLANG_DECLARE_SDK_IMPORTS_$$;
+      $$_INLANG_REASSIGN_SDK_IMPORTS_$$;
+    `),
+	)
+	return aliases
+}
+
+function replacePlaceholders(ast, config, fileType: "svelte" | "js") {
+	imports(
+		ast,
+		`@inlang/sdk-js/adapter-sveltekit/client/${config.languageInUrl ? "not-" : ""}reactive`,
+	).add("getRuntimeFromContext")
+	// Replace $$_INLANG_REASSIGN_SDK_IMPORTS_$$
+	identifiers(script.ast, "$$_INLANG_REASSIGN_SDK_IMPORTS_$$").replace(
+		parse(`
+      ({
+        ${aliases.map(({ exportN, aliasN }) => `${exportN}:${aliasN}`).join(",")}
+      } = getRuntimeFromContext());
+    `),
+	)
+	// Replace $$_INLANG_DECLARE_SDK_IMPORTS_$$
+	identifiers(script.ast, "$$_INLANG_DECLARE_SDK_IMPORTS_$$").replace(
+		parse(`
+      let ${aliases.map(({ exportN, aliasN }) => `${aliasN}`).join(",")}
+    `),
+	)
+	// Replace placeholders from sdk-js import, language
+	identifiers([script.ast, markup], "$$_INLANG_LANGUAGE_$$").replace(
+		parse(
+			!config.languageInUrl ? (fileType === "svelte" ? "$language" : "get(language)") : "language",
+		),
+	)
+	// Replace placeholders from sdk-js import, i
+	identifiers([script.ast, markup], "$$_INLANG_I_$$").replace(
+		parse(!config.languageInUrl ? (fileType === "svelte" ? "$i" : "get(i)") : "i"),
+	)
+	// Replace placeholders from sdk-js import, languages
+	identifiers([script.ast, markup], "$$_INLANG_LANGUAGES_$$").replace(
+		parse(
+			!config.languageInUrl
+				? fileType === "svelte"
+					? "$languages"
```

Comment:
`languages` is not a store

Reply:
Which exports are stores? only `i` and `language`, right?

Reply:
Yes..